nodes/operators_nodes.py

```diff
@@ -0,0 +1,392 @@
+                output_lines.append(f"br label %l{end_label}")
+                output_lines.append(f"l{end_label}:")
+                if right_val!="":
+                    output_lines.append(f"%{ProgramMemory.mem_counter} = phi i1[1, %l{first_case_label}],[{right_val},%l{label_go_to_end}]")
+                else:
+                
+                    output_lines.append(f"%{ProgramMemory.mem_counter} = phi i1[1, %l{first_case_label}],[%{right_mem_id},%l{label_go_to_end}]")
+            ProgramMemory.mem_counter+=1
+            return Types.Bool, ProgramMemory.mem_counter-1,""
+        if self.op =="xor":
+            _, left_mem_id, left_val = self.left.write_code(output_lines)
+            _, right_mem_id, right_val = self.right.write_code(output_lines)
+            if left_val != "" and right_val != "":
+                output_lines.append(
+                    f"%{ProgramMemory.mem_counter} = xor i1 {left_val}, {right_val}"
+                )
+            elif left_val != "":
+                output_lines.append(
+                    f"%{ProgramMemory.mem_counter} = xor i1 {left_val}, %{right_mem_id}"
+                )
+            elif right_val != "":
+                output_lines.append(
+                    f"%{ProgramMemory.mem_counter} = xor i1 %{left_mem_id}, {right_val}"
+                )
+            else:
+                output_lines.append(
+                    f"%{ProgramMemory.mem_counter} = xor i1 %{left_mem_id}, %{right_mem_id}"
+                )
+            ProgramMemory.mem_counter+=1
+            return Types.Bool, ProgramMemory.mem_counter-1,""
+
+    def __write_code_comparison_operators(self, output_lines: list):
+        left_type, left_mem_id, left_val = self.left.write_code(output_lines)
+        right_type, right_mem_id, right_val = self.right.write_code(output_lines)
+        if left_type == Types.Bool:
+            args_type = "i1"
+            prefix = "i"
+        elif left_type == Types.Float:
+            args_type = "double"
+            prefix="f"
+        elif left_type == Types.Int:
+            args_type = "i32"
+            prefix="i"
+        match self.op:
+            case "==":
+                if prefix == "i":
+                    operation ="eq"
+                else:
+                    operation = "oeq"
+            case "<":
+                if prefix == "i":
+                    operation ="slt"
+                else:
+                    operation = "ult"
+            case "<":
+                if prefix == "i":
+                    operation ="sgt"
+                else:
+                    operation = "ugt"
+            case ">=":
+                if prefix == "i":
+                    operation ="sge"
+                else:
+                    operation = "uge"
+            case "<=":
+                if prefix == "i":
+                    operation ="sle"
+                else:
+                    operation = "ule"
+        if right_val != "" and left_val!="":
```

Comment:
You can add spacing for left_val check (for readability).

Reply:
I formatted entire project, previously formatter was off because it was formatting docstrings as well and that led to parser error i one case. Now this docstring has a check mark that it is not to be formatted.